lib/ansible/modules/packaging/language/pip.py

```diff
@@ -503,12 +612,28 @@ def main():
         # Automatically apply -e option to extra_args when source is a VCS url. VCS
         # includes those beginning with svn+, git+, hg+ or bzr+
         has_vcs = False
+
         if name:
             for pkg in name:
-                if bool(pkg and re.match(r'(svn|git|hg|bzr)\+', pkg)):
+                if bool(pkg and _is_vcs_url(pkg)):
                     has_vcs = True
                     break
 
+            distributions = _get_distributions_from_names(module, name)
+            # check invalid combination of arguments
+            if version is not None:
+                if len(distributions) > 1:
+                    module.fail_json(
+                        msg="'version' argument is ambiguous when installing multiple package distributions. "
```

Comment:
I would call them `packages` instead of `distributions`.

Reply:
No, the correct name is distributions. In Python, packages are folders with python modules (`.py` files) and `__init__.py`, so that would be confusing terminology.
OTOH `package distributions` is a known compromise term for those doesn't really know about how python packaging works and its terminology.

Reply:
> It is common in the Python community to refer to a distribution using the term “package”. Using the term “distribution” is often not preferred, because it can easily be confused with a Linux distribution, or another larger software distribution like Python itself.
https://packaging.python.org/tutorials/installing-packages/

Reply:
"package" is more common in documentation and examples.

Reply:
Well, it was used to be known worse a while back. They've started working on that tutorial not long ago. Old users don't rush to reread those. That's why I prefer combo compromise to be as explicit as possible. Python packaging is in a weird state for a long time now. I don't want to add any more confusion.

Reply:
FWIW, +1 to package.

Distribution is only applicable to python, making it harder for people outside of that ecosystem to work on this module.

Reply:
Actually, even more than +1 to changing this to packages because this is in the error messages which we return to the user, not just about variable naming.  Everyone will understand what packages means in this context but sysadmins who just need to translate a script to setup the production environment their dev team gave them won't understand "distribution".